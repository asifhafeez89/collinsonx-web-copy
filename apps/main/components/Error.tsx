import { Notification, Text } from '@collinsonx/design-system/core';
import { ApolloError } from '@collinsonx/utils/apollo';

export interface ErrorProps {
  error?: ApolloError;
  title?: string;
}
const Error = ({ error, title = 'An error occurred' }: ErrorProps) => {
  return (
    <>
      {!!error ? (
        <Notification my={8} color="red.7" title={title} w="100%">
          {error.graphQLErrors.map((error, index) => (
            <Text key={index}>{error.message}</Text>
          ))}
        </Notification>
      ) : null}
    </>
  );
};

export default Error;
