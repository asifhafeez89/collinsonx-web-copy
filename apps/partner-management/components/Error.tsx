import { Text, Title } from '@collinsonx/design-system/core';
import { ApolloError } from '@collinsonx/utils/apollo';

export interface ErrorProps {
  error: ApolloError;
}
const Error = ({ error }: ErrorProps) => {
  return (
    <>
      <Title fw={600} order={3}>
        An error has occurred
      </Title>
      {error.graphQLErrors.map((error, index) => (
        <Text key={index}>{error.message}</Text>
      ))}
    </>
  );
};

export default Error;
