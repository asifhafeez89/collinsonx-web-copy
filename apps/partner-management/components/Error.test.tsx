import Error from './Error';
import '@testing-library/jest-dom/extend-expect';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  getNodeText,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';



describe('<Error />', () => {
    it('should render', () => {
  
      const component = render(
        <Error/>
      );
  
      expect(component).toMatchSnapshot();
    });
    it('should render with error', () => {
     let err = {name : 'error',
     message: 'there is some error',
     graphQLErrors:  [],
     protocolErrors: [],
     clientErrors: [],
     networkError: null,
     extraInfo: ''}
        const component = render(
          <Error
          error = {err}
          />
        );
    
        expect(component).toMatchSnapshot();
      });
}
)