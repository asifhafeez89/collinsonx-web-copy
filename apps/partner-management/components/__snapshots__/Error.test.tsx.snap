// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Error /> should render with error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  overflow: hidden;
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1.375rem;
  padding-right: 0.625rem;
  padding-top: 0.625rem;
  padding-bottom: 0.625rem;
  border-radius: 0.25rem;
  background-color: #fff;
  box-shadow: 0 0.0625rem 0.1875rem rgba(0, 0, 0, 0.05),rgba(0, 0, 0, 0.05) 0 1.75rem 1.4375rem -0.4375rem,rgba(0, 0, 0, 0.04) 0 0.75rem 0.75rem -0.4375rem;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-1::before {
  content: "";
  display: block;
  position: absolute;
  width: 0.375rem;
  top: 0.25rem;
  bottom: 0.25rem;
  left: 0.25rem;
  border-radius: 0.25rem;
  background-color: #f03e3e;
}

.emotion-1[data-with-icon] {
  padding-left: 0.625rem;
}

.emotion-1[data-with-icon]::before {
  display: none;
}

.emotion-1[data-with-border] {
  border: 0.0625rem solid #dee2e6;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  margin-right: 0.625rem;
}

.emotion-6 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: 0.875rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
  line-height: 1.4;
  margin-bottom: 0.125rem;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #212529;
}

.emotion-6:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-6:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: #868e96;
  font-size: 0.875rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #868e96;
  line-height: 1.4;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-9:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-9:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  position: relative;
  border-radius: 0.25rem;
  padding: 0;
  line-height: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 1.375rem;
  min-height: 1.375rem;
  width: 1.375rem;
  min-width: 1.375rem;
  border: 0.0625rem solid transparent;
  background-color: transparent;
  color: #868e96;
}

.emotion-14:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-14:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-14:hover {
    background-color: rgba(248, 249, 250, 1);
  }
}

@media (hover: none) {
  .emotion-14:active {
    background-color: rgba(248, 249, 250, 1);
  }
}

.emotion-14:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-14 [data-action-icon-loader] {
  max-width: 70%;
}

.emotion-14:disabled,
.emotion-14[data-disabled] {
  color: #ced4da;
  cursor: not-allowed;
  background-color: #f1f3f5;
  border-color: #f1f3f5;
  background-image: none;
  pointer-events: none;
}

.emotion-14:disabled:active,
.emotion-14[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-14[data-loading] {
  pointer-events: none;
}

.emotion-14[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

@media (hover: hover) {
  .emotion-14:hover {
    background-color: #f8f9fa;
  }
}

@media (hover: none) {
  .emotion-14:active {
    background-color: #f8f9fa;
  }
}

<body>
    <div>
      <div
        class="emotion-0 emotion-1"
        role="alert"
      >
        <div
          class="emotion-2 emotion-3"
        >
          <div
            class="emotion-4 emotion-5 emotion-6"
          >
            An error occurred
          </div>
          <div
            class="emotion-4 emotion-8 emotion-9"
          />
        </div>
        <button
          class="emotion-10 emotion-11 emotion-12 emotion-13 emotion-14"
          type="button"
        >
          <svg
            fill="none"
            style="width: 1rem; height: 1rem;"
            viewBox="0 0 15 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-1 {
  overflow: hidden;
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1.375rem;
  padding-right: 0.625rem;
  padding-top: 0.625rem;
  padding-bottom: 0.625rem;
  border-radius: 0.25rem;
  background-color: #fff;
  box-shadow: 0 0.0625rem 0.1875rem rgba(0, 0, 0, 0.05),rgba(0, 0, 0, 0.05) 0 1.75rem 1.4375rem -0.4375rem,rgba(0, 0, 0, 0.04) 0 0.75rem 0.75rem -0.4375rem;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-1::before {
  content: "";
  display: block;
  position: absolute;
  width: 0.375rem;
  top: 0.25rem;
  bottom: 0.25rem;
  left: 0.25rem;
  border-radius: 0.25rem;
  background-color: #f03e3e;
}

.emotion-1[data-with-icon] {
  padding-left: 0.625rem;
}

.emotion-1[data-with-icon]::before {
  display: none;
}

.emotion-1[data-with-border] {
  border: 0.0625rem solid #dee2e6;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  margin-right: 0.625rem;
}

.emotion-6 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: 0.875rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
  line-height: 1.4;
  margin-bottom: 0.125rem;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #212529;
}

.emotion-6:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-6:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: #868e96;
  font-size: 0.875rem;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #868e96;
  line-height: 1.4;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-9:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-9:focus:not(:focus-visible) {
  outline: none;
}

.emotion-14 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  position: relative;
  border-radius: 0.25rem;
  padding: 0;
  line-height: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 1.375rem;
  min-height: 1.375rem;
  width: 1.375rem;
  min-width: 1.375rem;
  border: 0.0625rem solid transparent;
  background-color: transparent;
  color: #868e96;
}

.emotion-14:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-14:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-14:hover {
    background-color: rgba(248, 249, 250, 1);
  }
}

@media (hover: none) {
  .emotion-14:active {
    background-color: rgba(248, 249, 250, 1);
  }
}

.emotion-14:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-14 [data-action-icon-loader] {
  max-width: 70%;
}

.emotion-14:disabled,
.emotion-14[data-disabled] {
  color: #ced4da;
  cursor: not-allowed;
  background-color: #f1f3f5;
  border-color: #f1f3f5;
  background-image: none;
  pointer-events: none;
}

.emotion-14:disabled:active,
.emotion-14[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-14[data-loading] {
  pointer-events: none;
}

.emotion-14[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

@media (hover: hover) {
  .emotion-14:hover {
    background-color: #f8f9fa;
  }
}

@media (hover: none) {
  .emotion-14:active {
    background-color: #f8f9fa;
  }
}

<div>
    <div
      class="emotion-0 emotion-1"
      role="alert"
    >
      <div
        class="emotion-2 emotion-3"
      >
        <div
          class="emotion-4 emotion-5 emotion-6"
        >
          An error occurred
        </div>
        <div
          class="emotion-4 emotion-8 emotion-9"
        />
      </div>
      <button
        class="emotion-10 emotion-11 emotion-12 emotion-13 emotion-14"
        type="button"
      >
        <svg
          fill="none"
          style="width: 1rem; height: 1rem;"
          viewBox="0 0 15 15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
